config/config.go

```diff
@@ -0,0 +1,145 @@
+package config
+
+import (
+	"fmt"
+	"github.com/fluent/fluent-bit-go/output"
+	"strconv"
+	"unsafe"
+)
+
+type PluginConfig struct {
+	BufferManagerConfig BufferConfig
+	NRClientConfig      NRClientConfig
+	ProxyConfig         ProxyConfig
+}
+
+type BufferConfig struct {
+	MaxBufferSize         int64
+	MaxRecords            int64
+	MaxTimeBetweenFlushes int64
+}
+
+type NRClientConfig struct {
+	Endpoint   string
+	ApiKey     string
+	LicenseKey string
+	UseApiKey  bool
+}
+
+type ProxyConfig struct {
+	IgnoreSystemProxy bool
+	Proxy             string
+	CABundleFile      string
+	CABundleDir       string
+	ValidateCerts     bool
+}
+
+func NewPluginConfig(ctx unsafe.Pointer) (cfg PluginConfig, err error) {
+	cfg.BufferManagerConfig, err = parseBufferConfig(ctx)
+	if err != nil {
+		return
+	}
+
+	cfg.NRClientConfig, err = parseNRClientConfig(ctx)
+	if err != nil {
+		return
+	}
+
+	cfg.ProxyConfig, err = parseProxyConfig(ctx)
+	if err != nil {
+		return
+	}
+
+	return cfg, nil
+}
+
+func parseBufferConfig(ctx unsafe.Pointer) (cfg BufferConfig, err error) {
+	maxBufferSize, err := optInt64(ctx, "maxBufferSize", 256000)
+	if err != nil {
+		return
+	}
+	cfg.MaxBufferSize = maxBufferSize
+
+	maxRecords, err := optInt64(ctx, "maxRecords", 1024)
+	if err != nil {
+		return
+	}
+	cfg.MaxRecords = maxRecords
+
+	maxTimeBetweenFlushes, err := optInt64(ctx, "maxTimeBetweenFlushes", 5000)
+	if err != nil {
+		return
+	}
+	cfg.MaxTimeBetweenFlushes = maxTimeBetweenFlushes
+
+	return cfg, err
+}
+
+func parseNRClientConfig(ctx unsafe.Pointer) (cfg NRClientConfig, err error) {
+	cfg.Endpoint = output.FLBPluginConfigKey(ctx, "endpoint")
+	if len(cfg.Endpoint) == 0 {
+		cfg.Endpoint = "https://log-api.newrelic.com/log/v1"
+	}
+	cfg.LicenseKey = output.FLBPluginConfigKey(ctx, "licenseKey")
+	cfg.ApiKey = output.FLBPluginConfigKey(ctx, "apiKey")
+
+	if len(cfg.ApiKey) == 0 && len(cfg.LicenseKey) == 0 {
+		err = fmt.Errorf("either apiKey or licenseKey must be specified")
+		return
+	}
+
+	if len(cfg.ApiKey) > 0 && len(cfg.LicenseKey) > 0 {
+		err = fmt.Errorf("only one of apiKey or licenseKey can be specified")
+		return
+	}
+
+	cfg.UseApiKey = len(cfg.ApiKey) > 0
+
+	return
```

Comment:
And this returns nothing, why the difference?  (I'm not super familiar with Go)

Reply:
Please see my comment above.